### PR TITLE
Add types for stats option's overload

### PR DESCRIPTION
**Pull request by olup**
This PR adds the stats return type when the stats option is enabled. It overseeds https://github.com/sindresorhus/globby/pull/173.

### CODE REVIEW

**Review by alex-page — APPROVED** [review]
Thanks @olup

**Comment by olup** [comment]
One question tho, is the stat option really returning objects of type `Stats` ?

**Comment by alex-page** [comment]
Yes because it has all of the different file stats if you select that value.

File: index.d.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import {Options as FastGlobOptions} from 'fast-glob';
+import {Stats} from 'fs';
 
 declare namespace globby {
 	type ExpandDirectoriesOption =
 		| boolean
 		| readonly string[]
-		| {files?: readonly string[]; extensions?: readonly string[]};
+		| { files?: readonly string[]; extensions?: readonly string[] };
```
Comment (sindresorhus) [inline index.d.ts:8]: Don't do unrelated changes.

File: index.d.ts [hunk 2]
```diff
@@ -86,12 +87,15 @@ declare const globby: {
...
 	@param patterns - See the supported [glob patterns](https://github.com/sindresorhus/globby#globbing-patterns).
 	@param options - See the [`fast-glob` options](https://github.com/mrmlnc/fast-glob#options-3) in addition to the ones in this package.
-	@returns The matching paths.
+	@returns The matching paths - or the matching path's stats if the `stats` option is set.
 	*/
-	sync: (
+	sync: ((
+		patterns: string | readonly string[],
+		options: globby.GlobbyOptions & { stats: true }
+	) => Stats[]) & ((
```
Comment (sindresorhus) [inline index.d.ts:95]: Why is this not just an overload?

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (olup) [reply index.d.ts:8]: Sorry, this is probably auto format. You don't seem to provide a prettier config or eslintrc ?

File: index.d.ts (hunk 2 above)
Comment (olup) [reply index.d.ts:95]: I am not sure but as this is declare as a key of a const object, you cannot just re-declare the key. There is a basic overload declared as the `globby` global result, but for the sync method, this is the best I could come up with.

File: index.d.ts (hunk 1 above)
Comment (sindresorhus) [reply index.d.ts:8]: > You don't seem to provide a prettier config or eslintrc ?

No, but you should default to not formatting if such configs do not exist.

File: index.d.ts (hunk 2 above)
Comment (sindresorhus) [reply index.d.ts:95]: Alright. You still need to fix the formatting.

Comment (olup) [reply index.d.ts:95]: Hum alright, I'll do that on Monday...

Comment (Purpzie) [reply index.d.ts:95]: Perhaps it could be changed into a regular overload? I'm unsure if it would be treated differently that way.

Apologies for butting in, I'm also editing this file for #177 and ran into this problem as well.

Comment (olup) [reply index.d.ts:95]: Sorry for the delay, better ?
